starknet_py/net/client.py

```diff
@@ -159,20 +157,13 @@ async def wait_for_tx(
             try:
                 tx_receipt = await self.get_transaction_receipt(tx_hash=tx_hash)
```

Comment:
maybe we could use `getTransactionStatus` here instead?

Reply:
The difference is that for `getTransactionStatus` field `execution_status` is optional, but for `getTransactionReceipt` both `execution_status` and `finality_status` are required. I think it might change the logic how this function works, so I would prefer not to change it.

Reply:
I think this is because `status` can be also used in not received/rejected txs that probably won't have execution_status (but don't quote me on that - let's double check). If that's the case, this is another reason to use status endpoint instead